pkg/i2gw/ingress2gateway.go

```diff
@@ -70,6 +83,73 @@ func Run(printer printers.ResourcePrinter, namespace string) {
 	outputResult(printer, httpRoutes, gateways)
 }
 
+func splitYAMLOrJSON(reader io.Reader) ([]*unstructured.Unstructured, error) {
+	d := kubeyaml.NewYAMLOrJSONDecoder(reader, 4096)
+	var objs []*unstructured.Unstructured
+	for {
+		u := &unstructured.Unstructured{}
+		if err := d.Decode(&u); err != nil {
+			if err == io.EOF {
+				break
+			}
+			return objs, fmt.Errorf("failed to unmarshal manifest: %v", err)
+		}
+		if u == nil {
+			continue
+		}
+		objs = append(objs, u)
+	}
+	return objs, nil
+}
+
+func readFile(l *networkingv1.IngressList, inputFile string) error {
```

Comment:
This function is doing much more than reading a file. Consider changing this to something like `constructIngressesFromFile`. 

Plus a comment on what it does would be good